controllers/packagevariants/pkg/controllers/packagevariant/packagevariant_controller.go

```diff
@@ -731,29 +730,30 @@ func (r *PackageVariantReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	//      we own, and use those to generate requests
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&api.PackageVariant{}).
-		Watches(&source.Kind{Type: &porchapi.PackageRevision{}},
-			handler.EnqueueRequestsFromMapFunc(r.mapObjectsToRequests)).
+		Watches(&porchapi.PackageRevision{}, handler.EnqueueRequestsFromMapFunc(mapObjectsToRequests(r.Client))).
 		Complete(r)
 }
 
-func (r *PackageVariantReconciler) mapObjectsToRequests(obj client.Object) []reconcile.Request {
-	attachedPackageVariants := &api.PackageVariantList{}
-	err := r.List(context.TODO(), attachedPackageVariants, &client.ListOptions{
-		Namespace: obj.GetNamespace(),
-	})
-	if err != nil {
-		return []reconcile.Request{}
-	}
-	requests := make([]reconcile.Request, len(attachedPackageVariants.Items))
-	for i, item := range attachedPackageVariants.Items {
-		requests[i] = reconcile.Request{
-			NamespacedName: types.NamespacedName{
-				Name:      item.GetName(),
-				Namespace: item.GetNamespace(),
-			},
+func mapObjectsToRequests(mgrClient client.Reader) handler.MapFunc {
+	return func(ctx context.Context, obj client.Object) []reconcile.Request {
+		attachedPackageVariants := &api.PackageVariantList{}
+		err := mgrClient.List(context.TODO(), attachedPackageVariants, &client.ListOptions{
```

Comment:
With the new api signature, you get the associated context as a param of the function, that should be used.

Reply:
Done